### DISCUSSION TITLE
"Select All" and "Do something" functionality within a collection

### DISCUSSION THREAD

**Discussion by vstarush** [payloadcms/payload#396] [Q&A]
Hi all,

I know this is probably solved by some custom component, but decided to ask before writing it all from scratch to see if maybe this problem was already solved or partially solved somewhere.

The problem:
For example, I have a collection "movies" and collection "genres". Within collection movies I want to **select all movies** (or filter them by name and select the result of the filter) and then add them to one of the genres in the collection "genres".

The implementation of genres is not really important. Essentially the functionality I am looking for is: **"Select All" / "Select filtered" => "Do something to all selected"**

Would appreciate any thoughts on this!

Thanks,
Vitaly

**Comment by DanRibbens** [comment]
Hi @vstarush!

This idea is something we have thought about as a future feature. I'm glad to see your post as it validates the idea further and raises it's importance. It could be done today with costom components and possible as a plugin with some extra engineering for reusabliity.

This capability belongs in the main product as it adds a lot of value and is difficult to bulid as a plug-in alone. I will see where we can place it in the roadmap and see. If you want to partner on the work for this, we potentially can co-author the feature if you are willing. Otherwise, we'll take a look at it in our development team and see where it fits in.

If anyone has a custom component already that might be helpful, please do share!

Thanks,
Dan

**Reply by vstarush** [reply]
Hi @DanRibbens!

Thank you for your answer and sorry for the long answer!

Co-authoring is a really cool suggestion, though this sounds like a significant commitment and I have some deadlines slowly creeping on me, so once I am done with those, if that is still an option, I would be more than glad!

I'll post something here again once I am back to solving this problem. Thank you again!

**Reply by DanRibbens** [reply]
I appreciate your enthusiasm @vstarush!

I'll update again when we have more info.
